From 20db3edfcb3bfaed60bfb5800cee3397cf550e8c Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:39:52 -0400
Subject: [virt] kvm/vmx: handle_dr does not handle RSP correctly

Message-id: <1406040016-3289-198-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86276
O-Subject: [RHEL7 PATCH v2 197/221] KVM: vmx: handle_dr does not handle RSP correctly
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Nadav Amit <namit@cs.technion.ac.il>

The RSP register is not automatically cached, causing mov DR instruction with
RSP to fail.  Instead the regular register accessing interface should be used.

Signed-off-by: Nadav Amit <namit@cs.technion.ac.il>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit a4ab9d0cf1ef0bf521bb69099aa464f38c71393c)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index a5fd47e..61e818d 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -5143,7 +5143,7 @@ static int handle_dr(struct kvm_vcpu *vcpu)
    return 1;
   kvm_register_write(vcpu, reg, val);
  } else
-  if (kvm_set_dr(vcpu, dr, vcpu->arch.regs[reg]))
+  if (kvm_set_dr(vcpu, dr, kvm_register_read(vcpu, reg)))
    return 1;
 
  skip_emulated_instruction(vcpu);
-- 
1.7.1